buildkite/bazelci.py

```diff
@@ -604,7 +629,15 @@ def execute_commands(
             raise BuildkiteException("working_directory refers to a path outside the workspace")
         os.chdir(requested_working_dir)
 
-        bazel_version = print_bazel_version_info(bazel_binary, platform)
+        # Resolve the specified version (e.g. "latest") to an actual version number (e.g. 0.22.0).
+        # If the binary is not a release (candidate), the result will be "unreleased binary".
+        resolved_bazel_version = print_bazel_version_info(bazel_binary, platform)
+        if bazel_version == "commit":
+            # Replace resolved_bazel_version with more precise information
+            # since we know the exact commit.
+            # However, we keep the call to print_bazel_version_info() above
+            # to get its output into the log.
+            resolved_bazel_version = "commit {}".format(os.environ.get("BUILDKITE_COMMIT"))
```

Comment:
When this script is running in a regular pipeline, the `BUILDKITE_COMMIT` is a commit of the downstream project, not a bazel commit.
Eg. https://buildkite.com/bazel/buildtools/builds/850#046743e5-a2b6-4bd1-a0f0-a992b610201a
BUILDKITE_COMMIT="4bcdbd1064fcc48180fa30400e39f7a940fdb8f9"

Reply:
Yeah, I didn't think this through properly - the build will probably also fail since the code might be missing. It's going to be more complex than anticipated :(

Reply:
To be more precise: The feature only makes sense in the one scenario that we wanted to handle (commits within the Bazel project), but not for other projects.

Reply:
A general approach would probably look like this:
if bazel:
  build bazel at commit # what we're doing in the current version of this PR
else:
  clone bazel
  build bazel at head